Makefile

```diff
@@ -4,3 +4,26 @@ design: ## render design diagrams
 	./docs/design/bin/render-plantuml.sh \
 		./docs/design/diagrams/build-sequence.plantuml \
 		./docs/design/diagrams/build-sequence.svg
+
+.PHONY: deny
+deny: ## check cargo deny lints
+	cargo deny --no-default-features check licenses bans sources
+
+.PHONY: clippy
+clippy: ## check cargo clippy lints
+	cargo clippy --locked -- -D warnings --no-deps
+
+.PHONY: fmt
+fmt: ## check cargo fmt lints
+	cargo fmt --check
+
+.PHONY: build
+build: ## build workspace in release mode
+	cargo build --release --locked
+
+.PHONY: test
+test: ## test workspace
+	cargo test --release --locked
+
+.PHONY: check
+check: fmt clippy build test
```

Comment:
It seems off to me for **build** to be included under **check**. Would it be better to remove **build** from there and add **check** as a pre-req of **build**? If someone wanted to build without the additional checks they can do a `cargo build` instead of a `make build`.